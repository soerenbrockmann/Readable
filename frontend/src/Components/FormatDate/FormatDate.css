.formatdate {
  margin-left: 0.5em;
}
