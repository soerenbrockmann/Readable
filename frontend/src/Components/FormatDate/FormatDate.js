import React from "react";
import "./FormatDate.css";

const FormatDate = props => {
  const { timestamp = Date.now() } = props;

  const options = {
    weekday: "long",
    year: "numeric",
    month: "long",
    day: "numeric",
    timeZone: "UTC",
    timeZoneName: "short"
  };

  const date = new Date(timestamp).toLocaleString("en-GB", options);
  return <div className="formatdate">{date}</div>;
};
export default FormatDate;
