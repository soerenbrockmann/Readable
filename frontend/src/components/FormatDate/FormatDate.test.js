import React from "react";
import { shallow } from "enzyme";
import FormatDate from "./FormatDate";

describe("<FormatDate />", () => {
  const timestamp = 1527512775719;
  const formatDate = shallow(<FormatDate timestamp={timestamp} />);

  it("renders properly", () => {
    expect(formatDate).toMatchSnapshot();
  });
});
