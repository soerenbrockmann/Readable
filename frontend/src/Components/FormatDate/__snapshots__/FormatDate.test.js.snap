// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormatDate /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormatDate
    timestamp={1527512775719}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Monday, May 28, 2018, UTC",
      "className": "formatdate",
    },
    "ref": null,
    "rendered": "Monday, May 28, 2018, UTC",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Monday, May 28, 2018, UTC",
        "className": "formatdate",
      },
      "ref": null,
      "rendered": "Monday, May 28, 2018, UTC",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
